Implement (Capital 'N') Normalization

CodaFi opened the pull request:
### What's in this pull request?

Add a clause-based normalization strategy and add a flag to allow the type checker to dump with it.  

This has revealed Bugs.

CodaFi committed:
Add complete normalization

CodaFi committed:
Add dumping of normalized metas

harlanhaskins commented on Sources/Mantle/Normalize.swift:
```diff
@@ -103,6 +141,79 @@ extension TypeChecker {
+            result = result.merge(self.matchClause(mergeEs, mergePats))
+            continue
+          default:
+            result = .failure(.fail(()))
```
Can you add a context for these failures?

CodaFi committed:
Properly lower closure bodies

An artifact of the old AST lowering: We used to drop the first
parameter because it would be scope checked as the first argument to
the reduce.  Now that we’re doing a proper right-fold here we need to
retain the argument

CodaFi committed:
Take inversion at its word when it reports blocking metas

CodaFi commented:
⛵️
